packages/oracle/src/dialect.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved
+
+import type { Sequelize } from '@sequelize/core';
+import { AbstractDialect } from '@sequelize/core';
+import type { SupportableNumericOptions } from '@sequelize/core/_non-semver-use-at-your-own-risk_/abstract-dialect/dialect.js';
+import { parseCommonConnectionUrlOptions } from '@sequelize/core/_non-semver-use-at-your-own-risk_/utils/connection-options.js';
+import { createNamedParamBindCollector } from '@sequelize/core/_non-semver-use-at-your-own-risk_/utils/sql.js';
+import {
+  BOOLEAN_CONNECTION_OPTION_NAMES,
+  CONNECTION_OPTION_NAMES,
+  NUMBER_CONNECTION_OPTION_NAMES,
+  STRING_CONNECTION_OPTION_NAMES,
+} from './_internal/connection-options.js';
+import * as DataTypes from './_internal/data-types-overrides';
+import { OracleConnectionManager } from './connection-manager';
+import type { OracleConnectionOptions, oracledbModule } from './connection-manager.js';
+import { OracleQueryGenerator } from './query-generator.js';
+import { OracleQueryInterface } from './query-interface.js';
+import { OracleQuery } from './query.js';
+
+export interface OracleDialectOptions {
+  /**
+   *  The oracledb module to user.
+   */
+  oracledbModule?: oracledbModule;
+}
+
+const numericOptions: SupportableNumericOptions = {
+  zerofill: false,
+  unsigned: true,
+};
+
+export class OracleDialect extends AbstractDialect<OracleDialectOptions, OracleConnectionOptions> {
+  static readonly supports = AbstractDialect.extendSupport({
+    'VALUES ()': true,
+    'LIMIT ON UPDATE': true,
+    lock: false,
+    forShare: 'LOCK IN SHARE MODE',
+    index: {
+      collate: false,
+      length: false,
+      parser: false,
+      type: false,
+      using: false,
+    },
+    constraints: {
+      restrict: false,
+      onUpdate: false,
+    },
+    returnValues: false,
+    returnIntoValues: true,
+    'ORDER NULLS': true,
+    schemas: true,
+    inserts: {
+      updateOnDuplicate: false,
+    },
+    indexViaAlter: false,
+    dataTypes: {
+      COLLATE_BINARY: true,
+      GEOMETRY: false,
+      JSON: true,
+      INTS: numericOptions,
+      DOUBLE: numericOptions,
+      DECIMAL: { unconstrained: true },
+      TIME: {
+        precision: false,
+      },
+    },
+    jsonOperations: true,
+    jsonExtraction: {
+      quoted: true,
+    },
+    dropTable: {
+      cascade: true,
+    },
+    renameTable: {
+      changeSchema: false,
+    },
+    delete: {
+      limit: true,
+    },
+    startTransaction: {
+      useBegin: true,
+    },
+    upserts: true,
+    bulkDefault: true,
+    topLevelOrderByRequired: true,
+  });
+
+  readonly connectionManager: OracleConnectionManager;
+  readonly queryGenerator: OracleQueryGenerator;
+  readonly queryInterface: OracleQueryInterface;
+  readonly Query = OracleQuery;
+  readonly dataTypesDocumentationUrl =
```

Comment:
this is passed to the parent constructor now, it's not a property you need to declare yourself anymore

Reply:
removed.